engine/interpreter/src/main/scala/pl/touk/nussknacker/engine/definition/TestInfoProvider.scala

```diff
@@ -63,9 +65,8 @@ class ModelDataTestInfoProvider(modelData: ModelData) extends TestInfoProvider {
   }
 
   private def contextWithOnlyGlobalVariables(implicit metaData: MetaData): ValidationContext = {
-    val globalTypes = modelData.processDefinition.expressionConfig.globalVariables.mapValues(_.returnType)
-    val typesWithMeta = MetaVariables.withType(globalTypes)
-    ValidationContext(Map.empty, typesWithMeta)
+    val globalTypes = globalVariablesPreparer.prepareGlobalVariables(metaData).mapValues(_.typ)
```

Comment:
Maybe you can create something like `prepareGlobalTypes(metadata)`? That will hide mapValues inside preparer.